test/checkers/id_checker.c

```diff
@@ -0,0 +1,61 @@
+#define LOG_PREFIX "TID_CHECKER: "
+#ifndef LOG_LEVEL
+    #define LOG_LEVEL INFO
+#endif
+#ifndef LOG_LOCKED
+    #define LOG_LOCKED
+#endif
+
+#include <dice/chains/capture.h>
+#include <dice/ensure.h>
+#include <dice/events/self.h>
+#include <dice/log.h>
+#include <dice/module.h>
+#include <dice/self.h>
+
+#if defined(__NetBSD__)
+     #include <lwp.h>
+static uint64_t
+osid_(void)
+{
+    return (uint64_t)_lwp_self();
+}
+#else
+static uint64_t
+osid_(void)
+{
+    return (uint64_t)gettid();
+}
+#endif
```

Comment:
I just noticed that there's no case for macos here. If this is intentional, I'm fine with it.